extension/storage/filestorage/extension.go

```diff
@@ -26,6 +26,19 @@ type localFileStorage struct {
 var _ storage.Extension = (*localFileStorage)(nil)
 
 func newLocalFileStorage(logger *zap.Logger, config *Config) (extension.Extension, error) {
+	if config.CreateDirectory {
+		var dirs []string
+		if config.Compaction.OnStart {
```

Comment:
I think you need to add OnRebound here as well?
```suggestion
		if config.Compaction.OnStart || config.Compaction.OnRebound {
```